### ISSUE TITLE
Change Maven group ID

### ISSUE THREAD

**Issue by robinst** [commonmark/commonmark-java#193]
After moving to the commonmark org (see #187), we should also change the group ID to `org.commonmark`. That will make it consistent with the Java package structure.

There's a guide here: https://frvabe.blogspot.com/2014/03/moving-maven-artifact-to-new-coordinates.html
And some more information here: https://maven.apache.org/guides/mini/guide-relocation.html

We also need to change the release process. Maybe it's a good idea to do another proper release to the old group ID first, and then do the relocation in a version that doesn't contain any new features.

**Comment by robinst** [comment]
Raised a ticket with Sonatype for the `org.commonmark` group ID: https://issues.sonatype.org/browse/OSSRH-62663

**Comment by robinst** [comment]
Done! Released 0.17.0 using groupId `org.commonmark`: https://search.maven.org/search?q=org.commonmark

And released a version of 0.17.0 under the old groupId as well but with a `relocation` section: https://search.maven.org/artifact/com.atlassian.commonmark/commonmark/0.17.0/jar

```
    <distributionManagement>
        <relocation>
            <groupId>org.commonmark</groupId>
        </relocation>
    </distributionManagement>
```

That should cause maven to print a warning to notify consumers of the new group ID when they run their build.
